ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/tabsontop/v1/tabsontop/clientlibs/site/js/tabs.js

```diff
@@ -326,15 +339,200 @@
+                this.#getClosestFieldsInView();
+            }
+        }
+
+        /**
+         * Adds unique HTML for added instance corresponding to requirements of different types of repeatableParent
+         * @param instanceManager of the repeated component
+         * @param addedModel of the repeated component
+         * @param htmlElement of the repeated component
+         */
+        addRepeatableMarkup(instanceManager,addedModel,htmlElement){
+            let result = this.#getBeforeViewElement(instanceManager, addedModel.index);
+            let beforeViewElement = result.beforeViewElement;
+            let elementToEnclose = result.elementToEnclose;
+            elementToEnclose.id = addedModel.id+result.idSuffix;
+            beforeViewElement.after(elementToEnclose);
+            elementToEnclose.append(htmlElement);
+        }
+
+
+        #cacheTemplateHTML(childView){
+            if(childView.getInstanceManager()!=null && (this._templateHTML==null || this._templateHTML[childView.getInstanceManager().getId()]==null)) {
+                var tabId = childView.element.id+"__tab";
+                var tabPanelId = childView.element.id+"__tabpanel";
+                var instanceManagerId = childView.getInstanceManager().getId();
+                var navigationTabToBeRepeated = this.#getTabNavElementById(tabId);
+                var tabPanelToBeRepeated = this.#getTabPanelElementById(tabPanelId);
+                this._templateHTML[instanceManagerId] = {};
+                this._templateHTML[instanceManagerId]['navigationTab'] = navigationTabToBeRepeated;
+                this._templateHTML[instanceManagerId]['targetTabPanel'] = tabPanelToBeRepeated;
+            }
+        }
+
+        #getClosestFieldsInView(){
+            var tabPanels=this.#getCachedTabPanels();
+            for(let i=0;i<tabPanels.length;i++){
+                var tabPanel=tabPanels[i];
+                var fieldView=this.getChild(tabPanel.id.substring(0,tabPanel.id.lastIndexOf("__")));
+                if(fieldView.getInstanceManager()!=null) {
+                    var instanceManagerId = fieldView.getInstanceManager().getId();
+                    if( this._templateHTML[instanceManagerId]['closestNonRepeatableFieldId'] ==null &&
+                        this._templateHTML[instanceManagerId]['closestRepeatableFieldInstanceManagerId'] ==null){
+                        var result=this.#getClosestFields(i);
+                        this._templateHTML[instanceManagerId]['closestNonRepeatableFieldId'] = result.closestNonRepeatableFieldId;
+                        this._templateHTML[instanceManagerId]['closestRepeatableFieldInstanceManagerId'] = result.closestRepeatableFieldInstanceManagerId;
+                    }
+                }
+            }
+        }
+
+
+
+        #getClosestFields(index){
+            var allTabPanels = this.#getCachedTabPanels();
+            var result={};
+            for (let i = 0; i < index; i++) {
```

Comment:
is there a need to iterate all tab panels since you are overriding the values in result?

Reply:
@nit23uec , we are iterating till the current tab panel in question. This is required to figure out the correct place to insert the added component in case all instances of the component were removed before that.